ui/helpers/utils/util.js

```diff
@@ -781,3 +781,46 @@ export const hexToText = (hex) => {
 export const getAvatarFallbackLetter = (subjectName) => {
   return subjectName?.match(/[a-z0-9]/iu)?.[0] ?? '?';
 };
+
+/**
+ * Get abstracted Snap permissions filtered by weight.
+ *
+ * @param weightedPermissions - Set of Snap permissions that have 'weight' property assigned.
+ * @param weightThreshold - Number that represents weight threshold for filtering.
+ * @param minPermissionCount - Minimum number of permissions to show,
+ * if filtered permissions count are less than the value specified.
+ * @returns Subset of permissions passing weight criteria.
+ */
+export const getFilteredSnapPermissions = (
+  weightedPermissions,
+  weightThreshold,
+  minPermissionCount = 3,
+) => {
+  const filteredPermissions = weightedPermissions.filter(
+    (permission) => permission.weight <= (weightThreshold ?? Infinity),
+  );
+
+  // If there are not enough permissions that fall into desired set filtered by weight,
+  // then fill the gap, no matter what the weight is
+  if (minPermissionCount && filteredPermissions.length < minPermissionCount) {
+    const remainingPermissions = [...weightedPermissions];
+
+    // Remove already filtered permissions to avoid duplicates
+    for (const permission of filteredPermissions) {
+      const index = remainingPermissions.indexOf(permission);
+      if (index > -1) {
+        remainingPermissions.splice(index, 1);
+      }
+    }
+
+    // Add permissions until desired count is reached
+    while (
+      filteredPermissions.length < minPermissionCount &&
+      remainingPermissions.length > 0
+    ) {
+      filteredPermissions.push(remainingPermissions.shift());
+    }
```

Comment:
```suggestion
    // Remove already included permissions to avoid duplicates
    const remainingPermissions = weightedPermissions.filter(
    (permission) => permission.weight > (weightThreshold ?? Infinity),
   );

    // Add permissions until desired count is reached
   filteredPermissions.concat(remainingPermissions.slice(0, minPermissionCount - filteredPermissions.length));
```

Perhaps slightly simpler like this?

Reply:
Yea, applied this proposal, but the only difference is that `filteredPermissions` needs to be reassigned, after concatenation.